### ISSUE TITLE
Either docs or actual issue with DNS

### ISSUE THREAD

**Issue by veera83372** [tailscale/tailscale#930]
**Describe the bug**

My setup is as follows:

LAN - 192.168.4.1/24
gateway - 192.168.4.1 (rapberrypi), 100.x.x.x(tailscale0)
I am running `dnsmasq` in my gateway.
Whenever I add Gateways's tailscale ip as dns server, all my public dns queries are not responding. 

```bash
pi@host:~/ $ # without tailscale DNS entries
pi@host:~/ $ cat /etc/resolv.conf
# Generated by resolvconf
nameserver 127.0.0.1
pi@host:~/ $ nslookup google.co.in
Server:		127.0.0.1
Address:	127.0.0.1#53

Non-authoritative answer:
Name:	google.co.in
Address: 142.250.67.131
Name:	google.co.in
Address: 2404:6800:4009:811::2003


```

```bash

pi@host:~/ $ # With tailscale internal machine and public DNS enteries
pi@host:~/ $ cat /etc/resolv.conf
# Generated by resolvconf
nameserver 100.x.x.x
nameserver 8.8.8.8

pi@host:~/ $ nslookup google.com
Server:		100.x.x.x
Address:	100.x.x.x#53

Non-authoritative answer:
Name:	google.com
Address: 216.58.199.142
Name:	google.com
Address: 2404:6800:4009:804::200e

```

```bash
pi@host:~/ $ # With only internal machine as DNS server
pi@host:~/ $ cat /etc/resolv.conf
# Generated by resolvconf
nameserver 100.x.x.x

pi@host:~/ $ nslookup google.com
Server:		100.x.x.x
Address:	100.x.x.x#53

** server can't find google.com: REFUSED
```

**To Reproduce**
1. Go to 'admin panel'
2. Add one of internal machine as DNS server and no public ones
3. Login to your machines 
4. See DNS resolution for public domains failed

**Expected behavior**
As it my DNS server capable of forwarding queries to root server, it still should be able to continue like this with out adding public one as fallback

**Screenshots**

**Version information:**
 - Device: raspberry pi
 - OS: Linux
 - OS version: Linux raspberrypi 5.4.72-v7+ #1356 SMP Thu Oct 22 13:56:54 BST 2020 armv7l GNU/Linux
 - Tailscale version: 
```1.2.6
  tailscale commit: 3f4e6d959a246369dab5866e21a2f7a0999cf715
  other commit: aad952c9af93ebb1d54a5ff2aee05190cac5d474
  go version: go1.15.4-tsf9db43b
```
**Additional context**

**Comment by llewxam-kache** [comment]
Are you sure dnsmasq is set to forward DNS requests to an upstream/external DNS server? 
But also, sounds like a similar issue to one I had, where I believe the DNS server points to itself as an 'external' resolver. 
https://github.com/tailscale/tailscale/issues/589#issuecomment-721310815

**Comment by veera83372** [comment]
Yes, dnsmasq is configured to forward DNS requests to an upstream, I am sure because all my LAN devices use it as DNS server. Yeah,  it is similar to your issue. May be we can tag this as duplicate

**Comment by danderson** [comment]
Should now be fixed, and will roll out in Tailscale 1.8. You can test-drive things with the unstable package on linux, or building from source on macOS/windows. We're still working on applying the polish, so you might hit rough edges, but the situation you ran into should no longer occur.
